modules/video/src/ecc.cpp

```diff
@@ -397,13 +398,26 @@ double cv::findTransformECC(InputArray templateImage,
     Mat templateFloat = Mat(hs, ws, CV_32F);// to store the (smoothed) template
     Mat imageFloat    = Mat(hd, wd, CV_32F);// to store the (smoothed) input image
     Mat imageWarped   = Mat(hs, ws, CV_32F);// to store the warped zero-mean input image
-    Mat allOnes		= Mat::ones(hd, wd, CV_8U); //to use it for mask warping
     Mat imageMask		= Mat(hs, ws, CV_8U); //to store the final mask
 
+    Mat inputMaskMat = inputMask.getMat();
+    //to use it for mask warping
+    Mat preMask =
+      inputMaskMat.empty() ? Mat::ones(hd, wd, CV_8U) : inputMaskMat;
+
     //gaussian filtering is optional
     src.convertTo(templateFloat, templateFloat.type());
     GaussianBlur(templateFloat, templateFloat, Size(5, 5), 0, 0);//is in-place filtering slower?
 
+    Mat preMaskFloat = Mat(hd, wd, CV_32F);
```

Comment:
no need to allocate matrix if it's output in the subsequent operations; just use "Mat preMaskFloat";